Sources/PrimerSDK/Classes/Core/Primer/Primer.swift

```diff
@@ -200,6 +210,117 @@ public class Primer {
+                                        let rvc = (UIApplication.shared.delegate as? UIApplicationDelegate)?.window??.rootViewController
+                                        
+                                        rvc?.dismiss(animated: true, completion: {
+                                            service.performChallenge(on: transaction, with: val, presentOn: rvc!, completion: { result in
+                                                switch result {
+                                                case .success(let netceteraAuthCompletion):
+                                                    let api: PrimerAPIClientProtocol = DependencyContainer.resolve()
+
+                                                    api.threeDSecurePostAuthentication(clientToken: clientToken, threeDSTokenId: paymentMethod.token!) { (result) in
+                                                        switch result {
+                                                        case .success(let data):
+                                                            completion(nil)
+                                                        case .failure(let err):
+                                                            completion(err)
+                                                        }
+                                                    }
+                                                    
+                                                case .failure(let err):
+                                                    completion(err)
+                                                }
+                                            })
+                                        })
+//                                                        let window = UIWindow(frame: UIScreen.main.bounds)
+//                                                        window.rootViewController = ClearViewController()
+//                                                        window.backgroundColor = UIColor.clear
+//                                                        window.windowLevel = UIWindow.Level.alert
+                                        
+                                        
+                                        
+                                    } else if let val = res?.authentication as? ThreeDS.BrowserV2ChallengeAPIResponse {
+                                        print(val)
+                                    } else if let val = res?.authentication as? ThreeDS.AppV2ChallengeAPIResponse {
+                                        print(val)
+                                    } else if let val = res?.authentication as? ThreeDS.BrowserV1ChallengeAPIResponse {
+                                        print(val)
+                                    } else if let val = res?.authentication as? ThreeDS.DeclinedAPIResponse {
+                                        print(val)
```

Comment:
[Q]
What's the logic for all these use cases?

Reply:
I believe that there's no logic here. I found those models when documentation existed (but then it was gone). I don't think I'll ever receive a BrowserV1ChallengeResponse. However, I'm waiting 3dsecure.io to provide us with more test cases so I can test those cases. They sent us a message that we'll have those today.